### DISCUSSION TITLE
Run picom with LightDM

### DISCUSSION THREAD

**Discussion by Monsterovich** [yshui/picom#1050] [Show and tell]
I was wondering if picom would work in LightDM. The `modesetting` driver only recently introduced TearFree, and without it everything without compositing shows screen tearing, including LightDM. Who cares about tearing in LightDM? I'm a perfectionist. :D

Create a file `/etc/lightdm/lightdm.conf` with the following content.

```ini
[SeatDefaults]
display-setup-script=$path_to_your_script$/lightdm.sh
```

lightdm.sh

```bash
#!/usr/bin/env bash

picom --daemon --config $path_to_config%/picom.conf
```

You need to explicitly specify the config, because lightdm is run by the root and therefore looks for the config in the root home folder.

That's it, goodbye tearing. The compositor will also run in a normal user session as if it was running when the DE started. At least in Xfce, it works.

Update:

You can also run picom with `--monitor-repaint` flag to check if it truly works. All refreshed parts of the screen will be displayed in red.

https://www.youtube.com/watch?v=VcyOHm_2ths

**Comment by ghost** [comment]
You can also check here which I just found from [arch wiki reference](https://wiki.archlinux.org/title/picom): https://ubuntuforums.org/showthread.php?t=2144468&p=12644745#post12644745
